src/all/comickfun/src/eu/kanade/tachiyomi/extension/all/comickfun/Comick.kt

```diff
@@ -399,9 +399,15 @@ abstract class Comick(
                 .parseAs<Covers>().mdCovers.reversed().toMutableList()
             if (covers.any { it.vol == "1" }) covers.retainAll { it.vol == "1" }
             if (
-                covers.any { it.locale == comickLang.split('-').first() }
+                covers.any { comickLang.startsWith(it.locale.orEmpty()) }
             ) {
-                covers.retainAll { it.locale == comickLang.split('-').first() }
+                covers.retainAll { comickLang.startsWith(it.locale.orEmpty()) }
+            } else if (
+                covers.any { mangaData.comic.isoLang.orEmpty().startsWith(it.locale.orEmpty()) }
+            ) {
+                covers.retainAll {
```

Comment:
Might be better to use immutable list + when.

Reply:
I couldn't figure out how a `when` would fit in this situation, but I wasn't happy with all the duplicated code so I split each into its own filter.

Reply:
Nice, the new solution seems overall better than what you would achieve with `when`.